lib/pages/view_all_page.dart

```diff
@@ -1,42 +1,11 @@
 import 'package:flutter/material.dart';
-import 'package:project_fomo/components/shared/vertical_event_listing.dart';
-import 'package:project_fomo/style.dart';
+import 'package:project_fomo/components/view_all/view_all_body.dart';
 
 class ViewAllPage extends StatelessWidget {
   static const String pageRoute = '/view-all';
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        leading: IconButton(
-          icon: Icon(
-            Icons.arrow_back_ios,
-            color: Colors.white,
-          ),
-          onPressed: () => Navigator.pop(context),
-        ),
-        title: Text(
-          'Popular',
-          style: TextStyle(
-            color: Colors.white,
-            fontSize: AppFontSize.size22,
-            fontFamily: AppFontFamily.family,
-          ),
-        ),
-        bottom: PreferredSize(
-          child: Container(
-            decoration: BoxDecoration(
-              gradient: AppColor.gradient,
-            ),
-            height: 2.0,
-          ),
-          preferredSize: Size.fromHeight(2.0),
-        ),
-        backgroundColor: Colors.transparent,
-        elevation: 0.0,
-      ),
-      body: VerticalEventListing(),
-    );
+    return ViewAllBody();
```

Comment:
View all body is unnecessary in this case.  Just make this page stateful and move the code here.

Reply:
Technically true but violates design. Make sure you want this. The current structure is page in our app has a "page" file under pages and a "body" component under components/[page]. If you think it is worth violating design it is not a hard change to make.

Reply:
I used that design in the case of needing to create and provide a bloc for a page at the lowest level possible.  So if a page needs a bloc to manage state, I would use provider to setup creation, disposal, and then pass a PageBody as a child so that the bloc is provided to the parts of the page that need to change/have state.  In this case you have the view all bloc much higher up, so you do not need to split the body off.

Reply:
Ok. Just committed a single stateful ViewAllPage. ViewAllBody has been removed. Note: components/viewall is now an empty directory but I kept it around for consistency